telos-frontend/src/components/todo/JournalTodo.jsx

```diff
@@ -1,3 +1,386 @@
+                >
+                  <AddIcon className="Publish" aria-controls="simple-modal" />
+                </IconButton>
+                <Dialog open={open} onClose={handleClose} aria-labelledby="form-dialog-title">
+                  <DialogTitle id="form-dialog-title">New To Do</DialogTitle>
+                  <DialogContent>
+                    <TextField
+                      autoFocus
+                      margin="dense"
+                      id="name"
+                      value={item.name}
+                      onChange={firstEvent}
+                      label="Description"
+                      fullWidth
+                    />
+                    <TextField
+                      id="date"
+                      label="Due Date"
+                      labelColour="black"
+                      type="date"
+                      defaultValue="2020-05-24"
+                      fullWidth
+                      InputLabelProps={{
+                        shrink: true,
+                      }}
+                    />
+                  </DialogContent>
+                  <DialogActions>
+                    <Button className={classes.button} onClick={handleClose}>
+                      Cancel
+                    </Button>
+                    <Button
+                      className={classes.button}
+                      label="Button"
+                      onClick={() => {
+                        setNewItem((prev) => [...prev, item], setItem(''), handleClose());
+                      }}
+                    >
+                      Confirm
+                    </Button>
+                  </DialogActions>
+                </Dialog>
+              </InputAdornment>
+            }
+            labelWidth={70}
+          />
+        </FormControl>
+      </div>
+      <Menu
+        id="simple-menu"
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        <MenuItem
+          className="menubar"
+          onClick={() => {
+            openMigrate();
+            setAnchorEl(null);
+          }}
+        >
+          Migrate
+        </MenuItem>
+        <MenuItem
+          className="menubar"
+          onClick={() => {
+            cancelEvent();
+            setAnchorEl(null);
+          }}
+        >
+          {cancel.indexOf(newItem[activeIndex]) !== -1 ? 'Uncancel' : 'Cancel'}
+        </MenuItem>
+        <MenuItem
+          id="delete"
+          className="menubar"
+          onClick={() => {
+            deleteEvent();
+            setAnchorEl(null);
+          }}
+        >
+          Delete
+        </MenuItem>
+      </Menu>
+      <Dialog open={migrate} onClose={closeMigrate} aria-labelledby="form-dialog-title">
+        <DialogTitle id="form-dialog-title">Migrate Event</DialogTitle>
+        <DialogContent>
+          <form noValidate>
+            <TextField
+              id="date"
+              label="Due Date"
```

Comment:
For the migrate, the user should be able to select the date for the task to move to, as well as the new due date, is it possible to include this? Otherwise, changing 'due date' to something like 'date to move to' would be good if that's possible, because migrating doesn't necessarily change the due date :)

Reply:
Added :)
![image](https://user-images.githubusercontent.com/50128761/111888460-99a27400-8a41-11eb-8fbd-e927829ca098.png)